wren-ui/src/apollo/server/resolvers/dashboardResolver.ts

```diff
@@ -5,21 +5,59 @@ import {
   PreviewDataResponse,
   DEFAULT_PREVIEW_LIMIT,
 } from '@server/services';
-import { DashboardItem, DashboardItemType } from '@server/repositories';
+import {
+  Dashboard,
+  DashboardItem,
+  DashboardItemType,
+} from '@server/repositories';
 import { getLogger } from '@server/utils';
+import {
+  SetDashboardCacheData,
+  DashboardSchedule,
+  PreviewItemResponse,
+} from '@server/models/dashboard';
 
 const logger = getLogger('DashboardResolver');
 logger.level = 'debug';
 
 export class DashboardResolver {
   constructor() {
+    this.getDashboard = this.getDashboard.bind(this);
     this.getDashboardItems = this.getDashboardItems.bind(this);
     this.createDashboardItem = this.createDashboardItem.bind(this);
     this.updateDashboardItem = this.updateDashboardItem.bind(this);
     this.deleteDashboardItem = this.deleteDashboardItem.bind(this);
     this.updateDashboardItemLayouts =
       this.updateDashboardItemLayouts.bind(this);
     this.previewItemSQL = this.previewItemSQL.bind(this);
+    this.setDashboardSchedule = this.setDashboardSchedule.bind(this);
+  }
+
+  public async getDashboard(
+    _root: any,
+    _args: any,
+    ctx: IContext,
+  ): Promise<
+    Omit<Dashboard, 'nextScheduledAt'> & {
+      schedule: DashboardSchedule;
+      items: DashboardItem[];
+      nextScheduledAt: string | null;
+    }
```

Comment:
Why do we omit `nextScheduledAt`, but define it again in the unioned type ?

Reply:
nextScheduledAt in `Dashboard` defined as `Date`, but we want to represent as `string` here.